### PR TITLE
Use tokenizer instead of copying index for `add`

### PR DESCRIPTION
**Pull request by genryxy**
Part of #113
Use reactive tokenizer for `add` instead of copying index file in temporary file. Now it reads line by line from source file and write a new index to a temporary file which at the end replaces source index file

### CODE REVIEW

File: src/main/java/com/artipie/helm/AddWriter.java [hunk 1]
```diff
@@ -131,62 +136,95 @@
...
+                        final BufferedWriter bufw = new BufferedWriter(
+                            new OutputStreamWriter(Files.newOutputStream(out))
+                        );
+                        final TokenizerFlatProc target = new TokenizerFlatProc("\n");
+                        final AtomicReference<String> name = new AtomicReference<>();
+                        final AtomicBoolean entrs = new AtomicBoolean(false);
+                        final AtomicReference<YamlWriter> writer = new AtomicReference<>();
+                        writer.set(new YamlWriter(bufw, 2));
+                        final AtomicReference<LineWriter> linewrtr = new AtomicReference<>();
+                        linewrtr.set(new LineWriter(writer.get()));
```
Comment (g4s8) [inline src/main/java/com/artipie/helm/AddWriter.java:155]: Here is the same problem as in another PR: global context variables are hard to read and understand (and they may contain concurrency bugs), please use some reducer to provide operation context

**Review by g4s8 — CHANGES_REQUESTED** [review]
@genryxy please check my comment

**Comment by genryxy** [comment]
@g4s8 thanks, I tried to simplify using `scan` as it was suggested. Also I noticed that class `LineWriter` is extra and could be simply included in class `YamlWriter`

**Comment by genryxy** [comment]
@g4s8 could you please take a look

**Comment by genryxy** [comment]
@g4s8 please check

**Review by g4s8 — APPROVED** [review]
@genryxy thanks

**Pull request merged**